SoftwareAG/cumulocity-clients-dotnet
# Updated sources based on review comments.

Implemented:
- Reduced overall warning count
- Added `CancellationToken` as optional parameter for each operation
- Use `using` with disposable objects
- Use `.ConfigureAwait(false)` to not force deadlocks
- Fixed conversion of Date objects
- Collections will not be optional and always declared empty
- Replace 3.1 and 6.0 clients with netstandard2.1
- Updated dependencies to use lastest versions
- Use file scoped namespaces
- Delete redundant conditional access
- Delete redundant #nullable
- Namespaces reflect folder structure
- Removed base class
- Extensions use their own file
- API/Test classes are sealed
- Simplified Obsolete
- Use constants
- URLEncoding
- Corrected file header comment
- Mark not extensible classes as sealed
- Refactored Converters and reduced warning count
- Factory to use `lazy` instantiation
- Factory based on interfaces
- Use `IDictionary` for custom fragments
- Serialization Wrapper
- Changed return type of resources returning `Task<System.IO.Stream>` to `Task<string?>`

Open
- Error handling
  - Discriminated union will follow - Need an option to access response headers
- Internal classes should not have the same type as external (type generics)
- Rework solution to handle readOnly properties

Not implemented
- `IReadOnlyList` might cause some issues, e.g. programmatically creating a collection
- `ReadOnlyListJsonConvertFactory` is not required. Collections will use non-null values even though keys are not present.
- APIs are by design not internal to keep track on different authorization settings, e.g. bootstrap
- Change api methods which returns `Task<System.IO.Stream>`
  - When `Accept` is `application/zip`, `application/octet-stream` return signature should be `Task<byte[]>`
  - Will be done later, resources returning binary data need to be treated differently
